src/main/java/org/candlepin/subscriptions/tally/collector/RHELProductUsageCollector.java

```diff
@@ -48,8 +49,9 @@ public Optional<HostTallyBucket> collect(
     // Cloud provider hosts only account for a single socket.
     if (normalizedFacts.getCloudProviderType() != null) {
       appliedSockets = normalizedFacts.isMarketplace() ? 0 : 1;
-      prodCalc.addCloudProvider(
```

Comment:
Would it make sense just to keep this method but redefine it as

```
add(normalizedFacts.getCloudProviderType(), Uom.CORES, (double) appliedCores);
add(normalizedFacts.getCloudProviderType(), Uom.SOCKETS, (double) appliedSockets);
add(normalizedFacts.getCloudProviderType(), Uom.INSTANCES, (double) 1)
```

I know the method is marked deprecated now, but we could just fix it and unmark it since it's only used internally.  To me it makes sense to keep those three `adds` together in one logical unit.  Or maybe just a convenience method like

```
public void addStandardMeasurementTuple(hardwareMeasurementType, cores, sockets, instances)
```

And then we could just use that method throughout this file and don't need to worry about future copy paste errors where someone misses one of the 3 critical `add`s.

Reply:
@awood, fixed, PTAL